ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -372,12 +371,17 @@ public void onCompletion(String result, Exception exception) {
               router.deleteBlob(result, deleteCallback);
               break;
             default:
-              throw new IllegalStateException("Unrecognized RestMethod: " + restMethod);
+              exception = new IllegalStateException("Unrecognized RestMethod: " + restMethod);
           }
+        } catch (Exception e) {
+          exception = e;
         }
-      } finally {
-        callbackTracker.markCallbackProcessingEnd();
       }
+
+      if (exception != null) {
+        submitResponse(restRequest, restResponseChannel, null, exception);
```

Comment:
The new code seems to directly submit response if there is an exception. What would be the benefit comparing with throwing an exception and let the caller to handle the exception?